src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataZipArchiveService.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Storage.Blob;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationUtils;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.FileTypeValidationUtils;
+using static GovUk.Education.ExploreEducationStatistics.Common.BlobContainerNames;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.FileStoragePathUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class DataZipArchiveService : IDataZipArchiveService
+    {
+        private readonly IFileTypeService _fileTypeService;
+
+        private static readonly Dictionary<ReleaseFileTypes, IEnumerable<Regex>> AllowedMimeTypesByFileType = 
+            new Dictionary<ReleaseFileTypes, IEnumerable<Regex>>
+            {
+                { ReleaseFileTypes.DataZip, AllowedZipMimeTypes }
+            };
+        
+        public DataZipArchiveService(IFileTypeService fileTypeService)
+        {
+            _fileTypeService = fileTypeService;
+        }
+
+        public async Task<Either<ActionResult, Tuple<ZipArchiveEntry, ZipArchiveEntry>>> GetArchiveEntries(
```

Comment:
Let's add some unit tests for this.

Reply:
Have added unit tests just for validating the archive file is valid & invalid for now.